src/java/org/apache/cassandra/net/InboundConnectionInitiator.java

```diff
@@ -533,26 +478,17 @@ void setupMessagingPipeline(InetAddressAndPort from, int useMessagingVersion, in
             {
                 case LZ4:
                 {
-                    if (useMessagingVersion >= VERSION_40)
-                        frameDecoder = FrameDecoderLZ4.fast(allocator);
-                    else
-                        frameDecoder = new FrameDecoderLegacyLZ4(allocator, useMessagingVersion);
+                    frameDecoder = FrameDecoderLZ4.fast(allocator);
```

Comment:
The comments for the `OutboundConnectionSettings.Framing` enum values have some references to the MS version that I think we don't need anymore: https://github.com/apache/cassandra/blob/530750f780b1b4a683263337574f2fa5cabd4dce/src/java/org/apache/cassandra/net/OutboundConnectionSettings.java#L53-L60